access/src/main/java/edu/unc/lib/dl/ui/controller/FedoraContentController.java

```diff
@@ -37,64 +53,94 @@
  */
 @Controller
 public class FedoraContentController {
-//    @Autowired
-//    private FedoraContentService fedoraContentService;
+    private static final Logger log = LoggerFactory.getLogger(FedoraContentController.class);
+
+    @Autowired
+    private FedoraContentService fedoraContentService;
+    @Autowired
+    private AnalyticsTrackerUtil analyticsTracker;
 
     @RequestMapping("/indexablecontent/{pid}")
     public void getDefaultIndexableContent(@PathVariable("pid") String pid,
             @RequestParam(value = "dl", defaultValue = "false") boolean download,
             HttpServletRequest request, HttpServletResponse response) {
-//        fedoraContentService.streamData(pid, ContentModelHelper.Datastream.DATA_FILE.getName(), download,
-//                new AnalyticsUserData(request), response);
+        streamData(pid, ORIGINAL_FILE, download, request, response);
     }
 
     @RequestMapping("/indexablecontent/{pid}/{datastream}")
     public void getIndexableContent(@PathVariable("pid") String pid, @PathVariable("datastream") String datastream,
             @RequestParam(value = "dl", defaultValue = "false") boolean download,
             HttpServletRequest request, HttpServletResponse response) {
-//        fedoraContentService.streamData(pid, datastream, download, new AnalyticsUserData(request), response);
+        streamData(pid, datastream, download, request, response);
     }
 
     @RequestMapping("/indexablecontent")
     public void getIndexableContentByParameters(@RequestParam("id") String id, @RequestParam("ds") String datastream,
             @RequestParam(value = "dl", defaultValue = "false") boolean download,
             HttpServletRequest request, HttpServletResponse response) {
-//        fedoraContentService.streamData(id, datastream, download, new AnalyticsUserData(request), response);
+        streamData(id, ORIGINAL_FILE, download, request, response);
     }
 
     @RequestMapping("/content/{pid}")
     public void getDefaultDatastream(@PathVariable("pid") String pid,
             @RequestParam(value = "dl", defaultValue = "false") boolean download,
             HttpServletRequest request, HttpServletResponse response) {
-//        fedoraContentService.streamData(pid, ContentModelHelper.Datastream.DATA_FILE.getName(), download,
-//                new AnalyticsUserData(request), response);
+        streamData(pid, ORIGINAL_FILE, download, request, response);
     }
 
     @RequestMapping("/content/{pid}/{datastream}")
     public void getDatastream(@PathVariable("pid") String pid, @PathVariable("datastream") String datastream,
```

Comment:
why are both getIndexableContent() and getDatastream() needed? they appear to behave identically other than their request mappings. ditto, for getIndexableContentByParameters and getDefaultDatastream(). maybe some javadoc would help clarify the purpose of the various methods :-)

Reply:
/indexablecontent/ is just for SEO so that we have a different path to identify files that we want search engines to download and index. For instance, we don't really want google to download our 26gb zip files, but we do want them to download and expose pdfs.

I'm actually not sure why I have them as separate controller methods since you can assign multiple paths to the same method, so I have refactored the class to only have a single set of methods.